lib/ASRouter.jsm

```diff
@@ -289,6 +305,28 @@ const MessageLoaderUtils = {
             provider.id,
             options.dispatchToAS
           );
+        } else if (["cfr", "cfr-fxa"].includes(provider.id)) {
```

Comment:
Is there anything `provider` specific about the file we download? Are we downloading the same file twice?

Reply:
RS downloader will compare the local file checksum against the remote one before sending out the download request. Nonetheless, you're right, it should just ignore the download if the previous one is already completed.

Reply:
I've added some basic mechanism to avoid the duplicate attachment downloads (even for attempts). Note that it only does so for one message load cycle, i.e. a call of `ASRouter.loadMessagesFromAllProviders()`, so that it will always check the freshness of the attachment for every message load.